src/main/java/com/joss/conductor/mobile/ConductorConfig.java

```diff
@@ -0,0 +1,379 @@
+    {
+        return platformName;
+    }
+
+    public String[] getCurrentSchemes()
+    {
+        return currentSchemes;
+    }
+
+    public String getFullAppPath() {
+        if(appFile == null)
+            return null;
+
+        Path path = Paths.get(appFile);
+        if(appFile.startsWith("sauce-storage:") || path.isAbsolute()) {
+            return appFile;
+        }
+
+        return Paths.get(System.getProperty("user.dir"), path.normalize().toString()).normalize().toString();
+    }
+
+    private void readConfig(InputStream is) {
+        Yaml yaml = new Yaml();
+        Map<String, Object> config = (Map<String, Object>)yaml.load(is);
+
+        String environmentPlatformName = System.getProperty("conductorPlatformName");
+        if(environmentPlatformName != null) {
+            platformName = Platform.valueOf(environmentPlatformName);
+        } else if(config.containsKey("platformName")) {
+            platformName = Platform.valueOf((String)config.get("platformName"));
+        }
+
+        Map<String, Object> defaults = (Map<String, Object>)config.get("defaults");
+        if(defaults != null) {
+            readProperties(defaults);
+
+            Map<String, Object> platformDefaults = null;
+            switch (platformName) {
+                case IOS:
+                    platformDefaults = (Map<String, Object>)defaults.get("ios");
+                    break;
+                case ANDROID:
+                    platformDefaults = (Map<String, Object>)defaults.get("android");
+                    break;
+            }
+            if(platformDefaults != null) {
+                readProperties(platformDefaults);
+            }
+        }
+
+        String environmentSchemes = System.getProperty("conductorCurrentSchemes");
+        if(environmentSchemes != null) {
+            currentSchemes = environmentSchemes.split(",");
+        } else {
+            List<String> schemesList = (List<String>)config.get("currentSchemes");
+            if(schemesList != null) {
+                currentSchemes = new String[schemesList.size()];
+                schemesList.toArray(currentSchemes);
+            }
+        }
+
+        if(currentSchemes != null) {
+
+            for(String scheme : currentSchemes) {
+                Map<String, Object> schemeData = (Map<String, Object>)config.get(scheme);
+                if(schemeData != null) {
+                    readProperties(schemeData);
+                }
+            }
+        }
+    }
+
+    private void readProperties(Map<String, Object> properties) {
+        for(String key : properties.keySet()) {
+            if(key.equals("ios") || key.equals("android")) {
+                // These keys are the start of platform specific options
+                // and should be ignored
+                continue;
+            }
+
+            setProperty(key, (String)properties.get(key).toString());
+        }
+    }
+
+    private void setProperty(String propertyName, String propertyValue) {
```

Comment:
I'm curious why you chose to write custom parsing instead of using a library like [jackson](https://github.com/FasterXML/jackson-dataformats-text/tree/master/yaml). Are we trying to avoid 3rd party libraries if possible?

Reply:
In this praticular case because the structure can have dynamic information at each area (and repeats). We're not straight parsing objects. Schemes, for example, can be anything, which Jackson doesn't really support well.